streams/src/test/java/org/apache/kafka/streams/query/PositionTest.java

```diff
@@ -0,0 +1,214 @@
+        assertThat(copy.getTopics(), equalTo(mkSet("topic", "topic1")));
+        assertThat(copy.getBound("topic"), equalTo(mkMap(mkEntry(0, 5L))));
+        assertThat(copy.getBound("topic1"), equalTo(mkMap(
+            mkEntry(0, 5L),
+            mkEntry(7, 0L)
+        )));
+
+        // original has changed
+        assertThat(position.getTopics(), equalTo(mkSet("topic", "topic1", "topic2")));
+        assertThat(position.getBound("topic"), equalTo(mkMap(mkEntry(0, 6L))));
+        assertThat(position.getBound("topic1"), equalTo(mkMap(
+            mkEntry(0, 5L),
+            mkEntry(7, 0L),
+            mkEntry(8, 1L)
+        )));
+        assertThat(position.getBound("topic2"), equalTo(mkMap(mkEntry(2, 4L))));
+    }
+
+    @Test
+    public void shouldMergeNull() {
+        final Position position = Position.fromMap(mkMap(
+            mkEntry("topic", mkMap(mkEntry(0, 5L))),
+            mkEntry("topic1", mkMap(
+                mkEntry(0, 5L),
+                mkEntry(7, 0L)
+            ))
+        ));
+
+        final Position merged = position.merge(null);
+
+        assertThat(merged.getTopics(), equalTo(mkSet("topic", "topic1")));
+        assertThat(merged.getBound("topic"), equalTo(mkMap(mkEntry(0, 5L))));
+        assertThat(merged.getBound("topic1"), equalTo(mkMap(
+            mkEntry(0, 5L),
+            mkEntry(7, 0L)
+        )));
+    }
+
+    @Test
+    public void shouldMatchOnEqual() {
+        final Position position1 = Position.emptyPosition();
+        final Position position2 = Position.emptyPosition();
+        position1.withComponent("topic1", 0, 1);
+        position2.withComponent("topic1", 0, 1);
+
+        position1.withComponent("topic1", 1, 2);
+        position2.withComponent("topic1", 1, 2);
+
+        position1.withComponent("topic1", 2, 1);
+        position2.withComponent("topic1", 2, 1);
+
+        position1.withComponent("topic2", 0, 0);
+        position2.withComponent("topic2", 0, 0);
+
+        assertEquals(position1, position2);
+    }
+
+    @Test
+    public void shouldNotMatchOnUnEqual() {
+        final Position position1 = Position.emptyPosition();
+        final Position position2 = Position.emptyPosition();
+        position1.withComponent("topic1", 0, 1);
+        position2.withComponent("topic1", 0, 1);
+
+        position1.withComponent("topic1", 1, 2);
+
+        position1.withComponent("topic1", 2, 1);
+        position2.withComponent("topic1", 2, 1);
+
+        position1.withComponent("topic2", 0, 0);
+        position2.withComponent("topic2", 0, 0);
+
+        assertNotEquals(position1, position2);
+    }
+
+    @Test
+    public void shouldNotMatchNull() {
+        final Position position = Position.emptyPosition();
+        assertNotEquals(position, null);
+    }
+
+    @Test
+    public void shouldMatchSelf() {
+        final Position position = Position.emptyPosition();
+        assertEquals(position, position);
+    }
+
+    @Test
+    public void shouldNotHash() {
```

Comment:
note: since this class is mutable, I wanted to make sure we don't accidentally use it in sets or anything.